app/reducers/companies.js

```diff
@@ -74,100 +75,65 @@ export type CompanyContactEntity = {
   id: number
 };
 
-function mutateCompanies(state, action) {
-  switch (action.type) {
-    case Company.DELETE.SUCCESS: {
-      return {
-        ...state,
-        items: state.items.filter(id => id !== action.meta.id)
-      };
-    }
-
-    case Company.ADD_SEMESTER_STATUS.SUCCESS: {
-      const companyId = action.meta.companyId;
-      const semesterStatuses = state.byId[companyId].semesterStatuses.concat(
-        action.payload
-      );
-      return mergeObjects(state, {
-        byId: {
-          [companyId]: { semesterStatuses }
-        }
-      });
-    }
-
-    case Company.EDIT_SEMESTER_STATUS.SUCCESS: {
-      const { companyId, semesterStatusId } = action.meta;
-      const semesterStatuses = state.byId[companyId].semesterStatuses.map(
-        status => (status.id === semesterStatusId ? action.payload : status)
-      );
-      return mergeObjects(state, {
-        byId: {
-          [companyId]: { semesterStatuses }
-        }
-      });
+type State = any;
+
+const mutateCompanies = produce(
+  (newState: State, action: any): void => {
+    switch (action.type) {
+      case Company.DELETE.SUCCESS:
+        pull(newState.items, action.meta.id);
```

Comment:
Hmm... Unsure about what I think about using utils doing in place edits on objects like this, since it makes my inner js-developer think that it doesn't change the object - even tho it does.

The alternative is
```
newState.items = without(newState.items, action.meta.id);
```

But I am not 100% sure about what i liked best.

Reply:
Yes, we can use without instead to be a bit more expressive.